### PR TITLE
Add ability to split intervals on band multiples

### PR DESCRIPTION
**Pull request by gbggrant**
Added functionality to take a list of intervals and split them on band multiples in order to allow for finer granularity of scattering.

### CODE REVIEW

File: src/java/htsjdk/samtools/util/IntervalList.java [hunk 1]
```diff
@@ -233,6 +233,60 @@ else if (current.intersects(next) || current.abuts(next)) {
...
+     */
+    private static List<Interval> breakIntervalAtBandMultiples(final Interval interval, final int bandMultiple) {
+        final List<Interval> brokenUpIntervals = new ArrayList<Interval>();
+
+        int startPos = interval.getStart();
+        int startIndex = startPos / bandMultiple;
+        final int endIndex = interval.getEnd() / bandMultiple;
+        while (startIndex <= endIndex) {
+            int endPos = (startIndex + 1) * bandMultiple -1;
+            if (endPos >= interval.getEnd()) {
```
Comment (yfarjoun) [inline src/java/htsjdk/samtools/util/IntervalList.java]: can be replaced with strict >

File: src/java/htsjdk/samtools/util/IntervalList.java [hunk 2]
```diff
@@ -233,6 +233,60 @@ else if (current.intersects(next) || current.abuts(next)) {
...
+     * @param intervals A list of Interval
+     * @param bandMultiple integer value (> 0) to break up intervals in the list at integer multiples of
+     * @return list of intervals that are broken up
+     */
+    public static List<Interval> breakIntervalsAtBandMultiples(final List<Interval> intervals, final int bandMultiple) {
+        final List<Interval> brokenUpIntervals = new ArrayList<Interval>();
+        for (final Interval i : intervals) {
+            final int startIndex = i.getStart() / bandMultiple;
+            final int endIndex = i.getEnd() / bandMultiple;
+            if (startIndex == endIndex) {
```
Comment (yfarjoun) [inline src/java/htsjdk/samtools/util/IntervalList.java]: Danger! End of an interval might be less than the start (empty intervals) seems to not account for that. (also below)

File: src/tests/java/htsjdk/samtools/util/IntervalListTest.java [hunk 3]
```diff
@@ -469,4 +470,33 @@ public void testMerges() {
         Assert.assertEquals(out.getStart(), 500);
         Assert.assertEquals(out.getEnd(), 699);
     }
+
+    @Test
+    public void testBreakAtBands() {
```
Comment (yfarjoun) [inline src/tests/java/htsjdk/samtools/util/IntervalListTest.java]: please also test for 
[start=11 end=10] with band == 10
[start=10 end=9] with band == 10

File: src/java/htsjdk/samtools/util/IntervalList.java (hunk 2 above)
Comment (gbggrant) [inline src/java/htsjdk/samtools/util/IntervalList.java]: Thanks - forgot about those.  Will empty intervals always be [start = n end = n-1] or could there be [start = n end = n - k]?

File: src/tests/java/htsjdk/samtools/util/IntervalListTest.java (hunk 3 above)
Comment (gbggrant) [inline src/tests/java/htsjdk/samtools/util/IntervalListTest.java]: Added those tests.  Actually:
101, 100 -> band == 100
100, 99 -> band == 100

**Comment by gbggrant** [comment]
Updated code to handle empty intervals.

File: src/java/htsjdk/samtools/util/IntervalList.java (hunk 2 above)
Comment (yfarjoun) [inline src/java/htsjdk/samtools/util/IntervalList.java]: always [n,n-1]

**Pull request merged**